packages/ember/tests/helpers/link_to_test/link_to_with_query_params_test.js

```diff
@@ -1,506 +1,577 @@
-  run(App, 'advanceReadiness');
-}
-
-
-function shouldNotBeActive(selector) {
-  checkActive(selector, false);
-}
-
-function shouldBeActive(selector) {
-  checkActive(selector, true);
-}
-
-function checkActive(selector, active) {
-  let classList = jQuery(selector, '#qunit-fixture')[0].className;
-  equal(classList.indexOf('active') > -1, active, selector + ' active should be ' + active.toString());
-}
-
-let updateCount, replaceCount;
-
-function sharedSetup() {
-  App = Application.create({
-    name: 'App',
-    rootElement: '#qunit-fixture'
-  });
-
-  App.deferReadiness();
-
-  updateCount = replaceCount = 0;
-  App.Router.reopen({
-    location: NoneLocation.create({
-      setURL(path) {
-        updateCount++;
-        set(this, 'path', path);
-      },
-
-      replaceURL(path) {
-        replaceCount++;
-        set(this, 'path', path);
-      }
-    })
-  });
-
-  Router = App.Router;
-  registry = App.__registry__;
-  container = App.__container__;
-}
-
-function sharedTeardown() {
-  run(() => App.destroy());
-  setTemplates({});
-}
-
-QUnit.module('The {{link-to}} helper: invoking with query params', {
-  setup() {
-    run(() => {
-      sharedSetup();
-
-      App.IndexController = Controller.extend({
-        queryParams: ['foo', 'bar', 'abool'],
-        foo: '123',
-        bar: 'abc',
-        boundThing: 'OMG',
-        abool: true
-      });
-
-      App.AboutController = Controller.extend({
+import { Route, Router } from 'ember-routing';
+import { moduleFor, ApplicationTestCase, AutobootApplicationTestCase } from 'internal-test-helpers';
+
+moduleFor('The {{link-to}} helper: invoking with query params', class extends ApplicationTestCase {
+  constructor() {
+    super();
+    let indexProperties = {
+      foo: '123',
+      bar: 'abc'
+    }
+    this.add('controller:index', Controller.extend({
+      queryParams: ['foo', 'bar', 'abool'],
+      foo: indexProperties.foo,
+      bar: indexProperties.bar,
+      boundThing: 'OMG',
+      abool: true
+    }));
+    this.add('controller:about', Controller.extend({
         queryParams: ['baz', 'bat'],
         baz: 'alex',
         bat: 'borf'
-      });
+    }));
+    this.indexProperties = indexProperties;
+  }
+
+  shouldNotBeActive(assert, selector) {
+    this.checkActive(assert, selector, false);
+  }
+
+  shouldBeActive(assert, selector) {
+    this.checkActive(assert, selector, true);
+  }
+
+  checkActive(assert, selector, active) {
+    let classList = this.$(selector)[0].className;
+    assert.equal(classList.indexOf('active') > -1, active, selector + ' active should be ' + active.toString());
+  }
+
+  [`@test doesn't update controller QP properties on current route when invoked`](assert) {
+    this.addTemplate('index', `
+      {{#link-to 'index' id='the-link'}}Index{{/link-to}}
+    `);
+
+    this.visit('/').then(() => {
+      this.$('#the-link').click();
+      let indexController = this.applicationInstance.lookup('controller:index');
```

Comment:
Why not extend `QueryParamTestCase` and make use of getController method here.

https://github.com/emberjs/ember.js/blob/master/packages/internal-test-helpers/lib/test-cases/query-param.js#L7
https://github.com/emberjs/ember.js/blob/master/packages/internal-test-helpers/lib/test-cases/query-param.js#L49

Or we can at least define getController method like that in here and make use of it.